builds/misc/templates/rust-test-module-build.yaml

```diff
@@ -18,14 +18,20 @@ steps:
     displayName: Build observability-agent client - amd64
     inputs:
       filePath: scripts/linux/cross-platform-rust-build.sh
-      arguments: --os alpine --arch amd64 --build-path test/modules/obsagent-client
+      arguments: >
+        --os alpine --arch amd64 --build-path test/modules/obsagent-client
+        --cargo-flags "--no-default-features --features prom"
   - task: Bash@3
     displayName: Build observability-agent client - arm32
     inputs:
       filePath: scripts/linux/cross-platform-rust-build.sh
-      arguments: --os ubuntu18.04 --arch arm32v7 --build-path test/modules/obsagent-client
+      arguments: >
+        --os ubuntu18.04 --arch arm32v7 --build-path test/modules/obsagent-client
+        --cargo-flags "--no-default-features --features prom"
   - task: Bash@3
     displayName: Build observability-agent client - arm64
     inputs:
       filePath: scripts/linux/cross-platform-rust-build.sh
-      arguments: --os ubuntu18.04 --arch aarch64 --build-path test/modules/obsagent-client
\ No newline at end of file
+      arguments: >
+        --os ubuntu18.04 --arch aarch64 --build-path test/modules/obsagent-client
+        --cargo-flags "--no-default-features --features prom"
```

Comment:
So we are choosing to have the pipeline output an image with only the prometheus feature? 

This module as a weird case as this configuration needs to be compilation based for resource reasons. What if we add the other features into the build pipeline also? This would increase the build image pipeline time so maybe it is best to build manually if we ever need different features. 

I am open to however you decide to deal with this.

Reply:
I changed it to build the image in dual mode (i.e. otel and prometheus).  As you suggested in your comment, there's the option to build in other modes manually.

Also, as discussed offline, eventually this obsagent-client module will reside in a separate repo with the obsagent. At that time we can decide if it's worth building images for all variants. For now, just building in dual mode.